tests/newCollectionDefinitions/books.js

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright 2023 Mia s.r.l.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict'
+
+const { STATES } = require('../../lib/consts')
+
+module.exports = {
+  name: 'books',
+  endpointBasePath: '/books-endpoint',
+  defaultState: 'DRAFT',
+  schema: {
+    type: 'object',
+    required: [
+      '_id',
+      'creatorId',
+      'createdAt',
+      'updaterId',
+      'updatedAt',
+      '__STATE__',
+      'name',
+      'isbn',
+    ],
+    properties: {
+      _id: {
+        type: 'string',
+        pattern: '^[a-fA-F0-9]{24}$',
+        specialType: 'ObjectId',
```

Comment:
Let's move this within a `__mia_configuration` object that allows to define custom properties specific for Mia
https://github.com/micro-lc/back-kit/blob/main/packages/engine/schemas/mia.schema.json#L51

Reply:
What if we move inside `__mia_config` also the other custom fields such as `nullable` and `encryption`?

Reply:
`nullable` property is already managed by Ajv to adhere OpenAPI 3 spec . On the contrary, it is a good idea to move also `encryption` property within `__mia_configuration` object.

Reply:
Done